src/hotspot/share/utilities/debug.cpp

```diff
@@ -181,16 +181,21 @@ void report_vm_error(const char* file, int line, const char* error_msg, const ch
 {
   va_list detail_args;
   va_start(detail_args, detail_fmt);
-  void* context = nullptr;
+
+  print_error_for_unit_test(error_msg, detail_fmt, detail_args);
+
+  const void* context = nullptr;
+  const void* siginfo = nullptr;
+
 #ifdef CAN_SHOW_REGISTERS_ON_ASSERT
-  if (g_assertion_context != nullptr && os::current_thread_id() == g_asserting_thread) {
-    context = g_assertion_context;
+  if (os::current_thread_id() == g_asserting_thread) {
+    context = os::get_saved_assert_context(&siginfo);
```

Comment:
I would leave the `g_assertion_context != nullptr` in. It should not happen, but its defensive and a bug at this point is difficult to figure out (because you just don't get a hs-err file)